source/mode/network_install_mode.cpp

```diff
@@ -195,16 +204,78 @@ namespace tin::ui
             LOG_DEBUG("%s", e.what());
             fprintf(stdout, "%s", e.what());
         }
+    }
 
-        if (m_clientSocket != 0)
+    void NetworkInstallMode::OnNSPSelected()
+    {
+        tin::ui::ViewManager& manager = tin::ui::ViewManager::Instance();
+        ConsoleCheckboxView* prevView;
+
+        if (!(prevView = dynamic_cast<ConsoleCheckboxView*>(manager.GetCurrentView())))
         {
-            close(m_clientSocket);
-            m_clientSocket = 0;
+            throw std::runtime_error("Previous view must be a ConsoleCheckboxView!");
         }
 
-        curl_global_cleanup();
-        #ifndef NXLINK_DEBUG
-        socketExit();
-        #endif
+        auto values = prevView->GetSelectedOptionValues();
+
+        for (auto& destStr : values)
+        {
+            m_urls.push_back(destStr->GetText());
+        }
+
+        auto view = std::make_unique<tin::ui::ConsoleOptionsView>(DEFAULT_TITLE);
+        view->AddEntry("Select Destination", tin::ui::ConsoleEntrySelectType::HEADING, nullptr);
+        view->AddEntry("", tin::ui::ConsoleEntrySelectType::NONE, nullptr);
+        view->AddEntry("SD Card", tin::ui::ConsoleEntrySelectType::SELECT, std::bind(&NetworkInstallMode::OnDestinationSelected, this));
+        view->AddEntry("NAND", tin::ui::ConsoleEntrySelectType::SELECT, std::bind(&NetworkInstallMode::OnDestinationSelected, this));
+        manager.PushView(std::move(view));
+    }
+
+    void NetworkInstallMode::OnDestinationSelected()
+    {
+        tin::ui::ViewManager& manager = tin::ui::ViewManager::Instance();
+        ConsoleOptionsView* prevView;
+
+        if (!(prevView = dynamic_cast<ConsoleOptionsView*>(manager.GetCurrentView())))
+        {
+            throw std::runtime_error("Previous view must be a ConsoleOptionsView!");
+        }
+
+        auto destStr = prevView->GetSelectedOptionValue()->GetText();
+        m_destStorageId = FsStorageId_SdCard;
+
+        if (destStr == "NAND")
+        {
+            m_destStorageId = FsStorageId_NandUser;
+        }
+
+        auto view = std::make_unique<tin::ui::ConsoleView>(4);
+        manager.PushView(std::move(view));
+
+        std::string nspExt = ".nsp";
+                    
+        for (auto& url : m_urls)
+        {
+            if (url.compare(url.size() - nspExt.size(), nspExt.size(), nspExt) == 0)
```

Comment:
This should actually be done earlier, probably about line 191. Otherwise users might be prompted to install files that aren't supported